src/render/src/device.rs

```diff
@@ -301,33 +297,41 @@ impl<B: Backend> Device<B> {
         })
     }
 
-    pub fn create_image_raw<A>(
+    pub fn create_image_raw<A, C>(
         &mut self,
         allocator: &mut A,
+        encoder: &mut Encoder<B, C>,
         usage: image::Usage,
         kind: image::Kind,
         mip_levels: image::Level,
         format: format::Format
     ) -> Result<handle::raw::Image<B>, image::CreationError>
-        where A: Allocator<B>
+        where A: Allocator<B>,
+              C: core::queue::Supports<core::Transfer>
```

Comment:
uh, yeah, that certainly looks out of place here. Device shouldn't know anything about queues and their capabilities